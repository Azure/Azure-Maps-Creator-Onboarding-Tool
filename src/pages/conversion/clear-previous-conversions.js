import { DefaultButton, Spinner, SpinnerSize } from '@fluentui/react';
import { clearCloudStorageData } from 'common/api/conversions';
import { useState } from 'react';
import toast from 'react-hot-toast';
import { actionButtonsRight } from './imdf-conversion.style';
import { logsButton } from './style';

export const ClearPreviousConversions = () => {
  const [loading, setLoading] = useState(false);

  const handleClick = async () => {
    setLoading(true);
    await clearCloudStorageData();
    setLoading(false);
    toast.success('Previous conversions deleted successfully');
  };

  return (
    <div className={actionButtonsRight}>
      {loading && <Spinner size={SpinnerSize.medium} />}
      <DefaultButton className={logsButton} onClick={handleClick} disabled={loading}>
        Delete Previous Conversions
      </DefaultButton>
    </div>
  );
};
