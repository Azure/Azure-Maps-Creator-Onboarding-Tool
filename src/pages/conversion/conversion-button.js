import { useCallback, useState, useEffect } from 'react';
import { cx } from '@emotion/css';
import { shallow } from 'zustand/shallow';
import { useTranslation } from 'react-i18next';

import { selectedStep as selectedStepStyle, step, stepTimer, stepTitle } from './style';
import Icon from './icon';
import { useConversionStore, conversionSteps } from 'common/store';
import { formatProgressTime } from './format-time';

const conversionStoreSelector = (s) => [s.conversionStepStatus, s.conversionStartTime, s.conversionEndTime, s.setStep, s.selectedStep];
let intervalId = null;

const ConversionButton = () => {
  const { t } = useTranslation();
  const [elapsedTime, setElapsedTime] = useState('');
  const [status, startTime, endTime, setStep, selectedStep] = useConversionStore(conversionStoreSelector, shallow);

  useEffect(() => {
    if (startTime !== null && endTime !== null) {
      setElapsedTime(formatProgressTime(startTime, endTime));
      clearInterval(intervalId);
    } else if (startTime !== null && intervalId === null) {
      intervalId = setInterval(() => {
        setElapsedTime(formatProgressTime(startTime, endTime));
      }, 1000);
    }

    return () => clearInterval(intervalId);
  }, [startTime, endTime]);

  const onClick = useCallback(() => setStep(conversionSteps.conversion), [setStep]);

  return (
    <button className={cx(step, { [selectedStepStyle]: selectedStep === conversionSteps.conversion })} onClick={onClick}
            aria-label={t('select.conversion.step')}>
      <div className={stepTitle}>
        <Icon status={status} />
        {t('package.conversion')}
      </div>
      <span className={stepTimer}>{elapsedTime}</span>
    </button>
  );
};

export default ConversionButton;