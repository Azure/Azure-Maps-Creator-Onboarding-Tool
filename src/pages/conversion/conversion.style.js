import { css } from '@emotion/css';

import { color, fontSize, fontWeight } from 'common/styles';

export const conversionContainer = css`
  display: flex;
  flex: 1;
  font-size: ${fontSize.sm};
  border-top: 1px solid ${color.gray};
`;

export const stepsContainer = css`
  width: 15rem;
  border-right: 1px solid ${color.gray};
`;

export const step = css`
  height: 2.75rem;
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 0 0.5rem;
  cursor: pointer;
  
  &:hover {
    color: ${color.accent.primary};
    background-color: ${color.lightGray};
  }
`;

export const selectedStep = css`
  background-color: ${color.gray};
  font-weight: ${fontWeight.bold};
`;

export const stepTitle = css`
  display: flex;
  align-items: center;
`;

export const successIcon = css`
  margin-right: 0.5rem;
  color: ${color.greenIcon};
`;

export const inProgressIcon = css`
  margin-right: 0.5rem;
  color: ${color.blueIcon};
`;

export const defaultIcon = css`
  margin-right: 0.5rem;
`;

export const failedIcon = css`
  margin-right: 0.5rem;
  color: ${color.redError};
`;

export const stepTimer = css`
  color: ${color.granite};
  font-size: ${fontSize.xxs};
`;

export const logsContainer = css`
  padding: 0 1rem;
`;