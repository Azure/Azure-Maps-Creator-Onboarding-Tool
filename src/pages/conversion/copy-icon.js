import { useState, useRef } from 'react';
import { Icon } from '@fluentui/react/lib/Icon';
import { mergeStyleSets, DelayedRender, Callout, Text, DirectionalHint } from '@fluentui/react';

import { copyIcon } from './style';

const styles = mergeStyleSets({
  callout: {
    padding: '0.25rem 0.5rem',
    borderRadius: 16,
    overflow: 'hidden',
    translate: '0px 0.3rem',
  },
});

const CopyIcon = ({textToCopy}) => {
  const [isCalloutVisible, setIsCalloutVisible] = useState(false);
  const ref = useRef(null);

  const copy = () => {
    navigator.clipboard.writeText(textToCopy);
    setIsCalloutVisible(true);
    setTimeout(() => {
      setIsCalloutVisible(false);
    }, 1000);
  };

  if (!textToCopy) {
    return null;
  }

  return (
    <>
      <span ref={ref}>
        <Icon iconName='Copy' onClick={copy} className={copyIcon} />
      </span>
      {isCalloutVisible && (
        <Callout target={ref} role='alert' directionalHint={DirectionalHint.topCenter} className={styles.callout}>
          <DelayedRender>
            <Text variant='small'>
              Copied!
            </Text>
          </DelayedRender>
        </Callout>
      )}
    </>
  );
};

export default CopyIcon;