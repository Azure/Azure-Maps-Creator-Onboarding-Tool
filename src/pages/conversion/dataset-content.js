import { shallow } from 'zustand/shallow';
import { useTranslation } from 'react-i18next';

import { conversionSteps, useConversionStore } from 'common/store';
import { logsContainer } from './style';

const conversionStoreSelector = (s) => [s.datasetOperationLog, s.datasetId, s.datasetOperationId, s.selectedStep];

const DatasetContent = () => {
  const { t } = useTranslation();
  const [datasetOperationLog, datasetId, datasetOperationId, selectedStep] = useConversionStore(conversionStoreSelector, shallow);

  if (selectedStep !== conversionSteps.dataset) {
    return null;
  }

  return (
    <div>
      <div>
        <h3>{t('meta.data')}</h3>
        <div>operationId: {datasetOperationId === null ? 'N/A' : datasetOperationId}</div>
        <div>datasetId: {datasetId === null ? 'N/A' : datasetId}</div>
      </div>
      <div>
        <h3>{t('operation.log')}</h3>
        <pre className={logsContainer}>{datasetOperationLog}</pre>
      </div>
    </div>
  );
};

export default DatasetContent;