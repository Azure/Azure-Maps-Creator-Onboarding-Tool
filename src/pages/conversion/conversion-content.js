import { shallow } from 'zustand/shallow';
import { useTranslation } from 'react-i18next';

import { conversionSteps, useConversionStore, conversionStatuses } from 'common/store/conversion.store';
import {
  boldHeader,
  contentContainer,
  logContainer,
  metaInfoContainer,
} from './style';
import { Log } from './log';
import { DownloadLogs } from './download-logs';
import CopyIcon from './copy-icon';

const conversionStoreSelector = (s) => [s.conversionStepStatus, s.conversionOperationLog, s.selectedStep, s.conversionId, s.diagnosticPackageLocation];

const ConversionContent = () => {
  const { t } = useTranslation();
  const [conversionStepStatus, operationLog, selectedStep, conversionId, diagnosticPackageLocation] = useConversionStore(conversionStoreSelector, shallow);

  if (selectedStep !== conversionSteps.conversion) {
    return null;
  }

  return (
    <div className={contentContainer}>
      <div className={metaInfoContainer}>
        <span className={boldHeader}>ConversionId</span>: {conversionId === null ? '' : conversionId}
        <CopyIcon textToCopy={conversionId} />
      </div>
      <div className={logContainer}>
        <h3>{t('operation.log')}</h3>
        <Log src={operationLog} />
      </div>
      {conversionStepStatus !== conversionStatuses.empty && conversionStepStatus !== conversionStatuses.inProgress &&
        <DownloadLogs type='conversion' isFailed={conversionStepStatus === conversionStatuses.failed}
                      link={diagnosticPackageLocation} json={operationLog} />}
    </div>
  );
};

export default ConversionContent;