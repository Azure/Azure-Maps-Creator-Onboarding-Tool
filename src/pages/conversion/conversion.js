import { conversionContainer, logsContainer, stepsContainer } from './conversion.style';
import UploadButton from './upload-button';
import UploadContent from './upload-content';

const Conversion = () => {
  return (
    <div className={conversionContainer}>
      <div className={stepsContainer}>
        <UploadButton selected />
      </div>
      <div className={logsContainer}>
        <UploadContent selected />
      </div>
    </div>
  );
};

export default Conversion;