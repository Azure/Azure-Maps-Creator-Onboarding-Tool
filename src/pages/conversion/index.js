import { shallow } from 'zustand/shallow';
import { useTranslation } from 'react-i18next';
import { useNavigate } from 'react-router-dom';
import { cx } from '@emotion/css';
import { Icon } from '@fluentui/react/lib/Icon';

import { PATHS } from 'common';
import {
  container,
  content,
  enabledStep,
  step as stepStyle,
  stepsContainer,
  stepTitle,
} from './style';
import UploadContent from './upload-content';
import ConversionContent from './conversion-content';
import DatasetContent from './dataset-content';
import TilesetContent from './tileset-content';
import MapContent from './map-content';
import StepButton from './step-button';
import { conversionSteps, useConversionStore } from 'common/store';
import { conversionStatuses } from 'common/store/conversion.store';

const conversionStoreSelector = (s) => [
  s.uploadStepStatus,
  s.uploadStartTime,
  s.uploadEndTime,
  s.conversionStepStatus,
  s.conversionStartTime,
  s.conversionEndTime,
  s.datasetStepStatus,
  s.datasetStartTime,
  s.datasetEndTime,
  s.tilesetStepStatus,
  s.tilesetStartTime,
  s.tilesetEndTime,
];

const Conversion = () => {
  const { t } = useTranslation();
  const navigate = useNavigate();

  const [
    uploadStepStatus,
    uploadStartTime,
    uploadEndTime,
    conversionStepStatus,
    conversionStartTime,
    conversionEndTime,
    datasetStepStatus,
    datasetStartTime,
    datasetEndTime,
    tilesetStepStatus,
    tilesetStartTime,
    tilesetEndTime,
  ] = useConversionStore(conversionStoreSelector, shallow);

  return (
    <div className={container}>
      <div className={stepsContainer}>
        <button onClick={() => navigate(PATHS.CONVERSIONS)} className={cx(stepStyle, enabledStep)}>
          <div className={stepTitle}>
            <Icon iconName='SkypeArrow' style={{marginRight: 8}} />
            {t('view.all.conversions')}
          </div>
        </button>
        <div style={{borderBottom: '2px solid'}}></div>
        <StepButton endTime={uploadEndTime} label={t('select.upload.step')} status={uploadStepStatus}
                    startTime={uploadStartTime} step={conversionSteps.upload} title={t('package.upload')} />
        <StepButton endTime={conversionEndTime} label={t('select.conversion.step')} status={conversionStepStatus}
                    startTime={conversionStartTime} step={conversionSteps.conversion} title={t('package.conversion')} />
        <StepButton endTime={datasetEndTime} label={t('select.dataset.step')} status={datasetStepStatus}
                    startTime={datasetStartTime} step={conversionSteps.dataset} title={t('dataset.creation')} />
        <StepButton endTime={tilesetEndTime} label={t('select.tileset.step')} status={tilesetStepStatus}
                    startTime={tilesetStartTime} step={conversionSteps.tileset} title={t('tileset.creation')} />
        <StepButton label={t('select.map')} status={tilesetStepStatus} step={conversionSteps.map} title={t('map')}
                    disabled={tilesetStepStatus !== conversionStatuses.finishedSuccessfully} />
      </div>
      <div className={content}>
        <UploadContent />
        <ConversionContent />
        <DatasetContent />
        <TilesetContent />
        <MapContent />
      </div>
    </div>
  );
};

export default Conversion;