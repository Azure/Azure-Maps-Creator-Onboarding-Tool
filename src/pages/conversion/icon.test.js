import { render } from '@testing-library/react';

import Icon from './icon';
import { conversionStatuses } from 'common/store/conversion.store';

describe('conversion icon', () => {
  it('should render success icon', () => {
    const view = render(<Icon status={conversionStatuses.finishedSuccessfully} />);
    expect(view).toMatchSnapshot();
  });

  it('should render inProgress icon', () => {
    const view = render(<Icon status={conversionStatuses.inProgress} />);
    expect(view).toMatchSnapshot();
  });

  it('should render failed icon', () => {
    const view = render(<Icon status={conversionStatuses.failed} />);
    expect(view).toMatchSnapshot();
  });

  it('should render default icon when status is "empty"', () => {
    const view = render(<Icon status={conversionStatuses.empty} />);
    expect(view).toMatchSnapshot();
  });

  it('should render default icon when status is not set', () => {
    const view = render(<Icon />);
    expect(view).toMatchSnapshot();
  });
});