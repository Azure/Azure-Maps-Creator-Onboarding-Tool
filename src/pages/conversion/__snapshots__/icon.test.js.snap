// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`conversion icon should render default icon when status is "empty" 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .emotion-0 {
  margin-right: 0.5rem;
}

<body>
    <div>
      <i
        aria-label="package.conversion"
        class="emotion-0 root-115"
        data-icon-name="StatusCircleRing"
        role="img"
      >
        
      </i>
    </div>
  </body>,
  "container": .emotion-0 {
  margin-right: 0.5rem;
}

<div>
    <i
      aria-label="package.conversion"
      class="emotion-0 root-115"
      data-icon-name="StatusCircleRing"
      role="img"
    >
      
    </i>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`conversion icon should render default icon when status is not set 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .emotion-0 {
  margin-right: 0.5rem;
}

<body>
    <div>
      <i
        aria-label="package.conversion"
        class="emotion-0 root-115"
        data-icon-name="StatusCircleRing"
        role="img"
      >
        
      </i>
    </div>
  </body>,
  "container": .emotion-0 {
  margin-right: 0.5rem;
}

<div>
    <i
      aria-label="package.conversion"
      class="emotion-0 root-115"
      data-icon-name="StatusCircleRing"
      role="img"
    >
      
    </i>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`conversion icon should render failed icon 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .emotion-0 {
  margin-right: 0.5rem;
  color: #A80000;
}

<body>
    <div>
      <i
        aria-label="package.conversion"
        class="emotion-0 root-112"
        data-icon-name="StatusErrorFull"
        role="img"
      >
        
      </i>
    </div>
  </body>,
  "container": .emotion-0 {
  margin-right: 0.5rem;
  color: #A80000;
}

<div>
    <i
      aria-label="package.conversion"
      class="emotion-0 root-112"
      data-icon-name="StatusErrorFull"
      role="img"
    >
      
    </i>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`conversion icon should render inProgress icon 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": @keyframes animation-0 {
  from {
    -webkit-transform: rotate(0deg);
    -moz-transform: rotate(0deg);
    -ms-transform: rotate(0deg);
    transform: rotate(0deg);
  }

  to {
    -webkit-transform: rotate(360deg);
    -moz-transform: rotate(360deg);
    -ms-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}

.emotion-0 {
  margin-right: 0.5rem;
  color: #0078d4;
  -webkit-animation: animation-0 2s linear infinite;
  animation: animation-0 2s linear infinite;
}

<body>
    <div>
      <i
        aria-label="package.conversion"
        class="emotion-0 root-109"
        data-icon-name="SyncStatusSolid"
        role="img"
      >
        
      </i>
    </div>
  </body>,
  "container": @keyframes animation-0 {
  from {
    -webkit-transform: rotate(0deg);
    -moz-transform: rotate(0deg);
    -ms-transform: rotate(0deg);
    transform: rotate(0deg);
  }

  to {
    -webkit-transform: rotate(360deg);
    -moz-transform: rotate(360deg);
    -ms-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}

.emotion-0 {
  margin-right: 0.5rem;
  color: #0078d4;
  -webkit-animation: animation-0 2s linear infinite;
  animation: animation-0 2s linear infinite;
}

<div>
    <i
      aria-label="package.conversion"
      class="emotion-0 root-109"
      data-icon-name="SyncStatusSolid"
      role="img"
    >
      
    </i>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`conversion icon should render success icon 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .emotion-0 {
  margin-right: 0.5rem;
  color: #107c10;
}

<body>
    <div>
      <i
        aria-label="package.conversion"
        class="emotion-0 root-106"
        data-icon-name="SkypeCircleCheck"
        role="img"
      >
        
      </i>
    </div>
  </body>,
  "container": .emotion-0 {
  margin-right: 0.5rem;
  color: #107c10;
}

<div>
    <i
      aria-label="package.conversion"
      class="emotion-0 root-106"
      data-icon-name="SkypeCircleCheck"
      role="img"
    >
      
    </i>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
